ext/random/engine_mt19937.c

```diff
@@ -279,8 +279,8 @@ PHP_METHOD(Random_Engine_Mt19937, __construct)
 
 	if (seed_is_null) {
 		/* MT19937 has a very large state, uses CSPRNG for seeding only */
-		if (php_random_bytes_silent(&seed, sizeof(zend_long)) == FAILURE) {
-			zend_throw_exception(spl_ce_RuntimeException, "Random number generation failed", 0);
+		if (php_random_bytes_throw(&seed, sizeof(zend_long)) == FAILURE) {
+			zend_throw_exception(spl_ce_RuntimeException, "Failed to generate a random seed", 0);
```

Comment:
`php_random_bytes_throw` will already have thrown an exception by now, and therefore this one should be eliminated.

Reply:
This one is intentionally preserved, as having both exceptions is a value-add:

1. The exception of php_random_bytes_throw() gives a *low level* explanation why exactly your runtime environment is broken (e.g. you are running an ancient kernel without getrandom() and you're also running in a chroot() that does not make `/dev/urandom` available.

2. This “Failed to generate a random seed” exception is a *high level* explanation of why you are seeing an exception here.

Consider this example with an intentionally broken CSPRNG:

```
<?php

try {
	new \Random\Engine\Xoshiro256StarStar();
} catch (\Exception $e) {
	echo $e->getMessage(), PHP_EOL;
	while ($e = $e->getPrevious()) {
		echo "caused by: ", $e->getMessage(), PHP_EOL;
	}
}
```

This will print:

> Failed to generate a random seed
> caused by: Cannot open source device

which is arguably more useful than just “Cannot open source device” (though this message could certainly be improved).

Reply:
> but I wonder whether we really want to throw a `RuntimeException` here.

No, we don't. This PR is part of my small step-by-step journey of cleaning up the error handling. See: https://github.com/php/php-src/pull/9071#issuecomment-1193283499

Reply:
@kocsismate Given cmb's answer, is this clear to merge from your side?